### ISSUE TITLE
Removing empty lines in a class can remove all errors from that file

### ISSUE THREAD

**Issue by Tragetaschen** [dotnet/roslyn#65209]
**Version Used**: 
17.4.0 Preview 6.0

**Steps to Reproduce**:

I have a small example C# file in my normal project. This file has errors (unimplemented interface and missing `async`):

```csharp
internal class Test : IEnumerable<int>
{
    public Task Method()
    {
        await Task.Delay(1000);
    }
}
```

I add a couple of empty lines in that file above the `Method` and then remove them randomly using Shift+Del.

**Expected Behavior**:
Nothing weird should happen.

**Actual Behavior**:
Each time I remove lines, one of three things that can happen in addition to the lines being gone:
1. Nothing else
2. Briefly, all red squiggles are gone and then come back
3. All red squiggles vanish and don't come back until the next edit

The gif starts with 2. then shows a couple of 1. and ends with a 3.
![red-squiggles-gone](https://user-images.githubusercontent.com/1947968/199676384-406ee2fe-d36b-4864-b8c0-7e2e8b7e8c94.gif)

**Comment by Youssef1313** [comment]
Likely related to https://github.com/dotnet/roslyn/pull/64885

**Comment by Tragetaschen** [comment]
Given the title of the linked PR: In my example, I never actually built the project. All errors only come from IntelliSense

**Comment by Youssef1313** [comment]
Yes, the diagnostics that are cleared in your case don't seem to be build-only diagnostics. But it needs investigation to know whether this is caused by #64885 (unintentionally) or not.

I linked this PR because it's a recent seemingly related change.

cc @mavasani

**Comment by mavasani** [comment]
@Youssef1313 https://github.com/dotnet/roslyn/pull/64885 shouldn't be causing this. I am seeing similar slowness in diagnostic computation in latest intpreview builds. Seems like an unrelated regression. I'll investigate

**Comment by mavasani** [comment]
@Tragetaschen Can you please confirm if the behavior you are experience is similar to https://github.com/dotnet/roslyn/issues/65502, i.e. error list refreshes at a reasonable rate and has up-to-date diagnostics, but the squiggles in the editor are slow to update or stale?

**Comment by Tragetaschen** [comment]
I have just checked in 17.4.0.

It reproduces and the squiggles vanish, but the error list empties as well.
The order is "squiggles vanish" and a couple of milliseconds later "error list empties".

![image](https://user-images.githubusercontent.com/1947968/203051982-7267f8f9-3cf3-4f6f-871d-99f9962ee644.png)

**Comment by mavasani** [comment]
@Tragetaschen I tried this a bunch on our latest internal 17.6 dogfood builds and I couldn't repro it even once: `Version 17.6.0 Preview 1.0 [33324.259.main]`.

Are you able to try it on the latest 17.5 public preview from https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-notes-preview and see if you can still reproduce it?
